In [1]:
import pandas as pd

In [2]:
sales = pd.read_csv("sales.csv")

In [4]:
%%sql
SELECT
    product_line,
    warehouse,
    client_type,
    SUM(quantity) AS total_quantity
FROM
    sales
GROUP BY
    product_line, warehouse, client_type
ORDER BY total_quantity DESC
LIMIT 10;

product_line warehouse client_type  total_quantity
0        Breaking system   Central   Wholesale           604.0
1  Suspension & traction   Central      Retail           508.0
2  Suspension & traction   Central   Wholesale           472.0
3  Suspension & traction     North   Wholesale           472.0
4           Frame & body     North   Wholesale           456.0
5      Electrical system   Central   Wholesale           436.0
6        Breaking system   Central      Retail           420.0
7      Electrical system   Central      Retail           402.0
8  Suspension & traction     North      Retail           394.0
9           Frame & body   Central   Wholesale           388.0

In [5]:
%%sql
SELECT
    CORR(unit_price, quantity) AS correlation_coefficient
FROM
    sales;

correlation_coefficient
0                 0.035822

In [9]:
%%sql
SELECT
    unit_price,
    SUM(quantity) AS total_quantity,
    SUM(total) AS total_revenue
FROM
    sales
GROUP BY
    unit_price
ORDER BY total_revenue DESC
LIMIT 10;

unit_price  total_quantity  total_revenue
0       63.66            40.0        2546.33
1       58.10            40.0        2324.20
2       63.41            36.0        2282.82
3       62.02            36.0        2232.66
4       60.73            36.0        2186.16
5       28.57            68.0        1942.74
6       46.15            40.0        1846.13
7       57.10            32.0        1827.04
8       39.50            46.0        1816.89
9       49.58            36.0        1784.88

# Quantity and Pricing Analysis

This report provides an analysis of the relationship between quantity ordered and unit pricing based on the provided dataset.

## Overview

The Quantity and Pricing Analysis aims to understand how unit pricing affects the quantity of products ordered and the total revenue generated. By examining pricing strategies and their impact on sales volume and revenue, businesses can optimize their pricing strategies to maximize profitability.

## Findings

### Top Pricing Strategies

Based on the dataset, the top 10 pricing strategies, along with their corresponding total quantities ordered and total revenue generated, are as follows:

| Unit Price | Total Quantity | Total Revenue |
|------------|----------------|---------------|
| 63.66      | 40             | $2546.33      |
| 58.10      | 40             | $2324.20      |
| 63.41      | 36             | $2282.82      |
| 62.02      | 36             | $2232.66      |
| 60.73      | 36             | $2186.16      |
| 28.57      | 68             | $1942.74      |
| 46.15      | 40             | $1846.13      |
| 57.10      | 32             | $1827.04      |
| 39.50      | 46             | $1816.89      |
| 49.58      | 36             | $1784.88      |

### Observations

- There is a range of unit prices with varying total quantities ordered and total revenue generated.
- Higher unit prices do not always result in higher total revenue, as seen in some cases where lower-priced items generate more revenue.
- Similarly, higher total quantities ordered do not necessarily correlate with higher total revenue, as seen with some lower total quantities associated with higher total revenue.

### Pricing Strategies

- **Higher Price, Higher Revenue**: Products with higher unit prices generate substantial revenue despite relatively lower quantities ordered.
- **Lower Price, Higher Volume**: Products with lower unit prices have higher total quantities ordered, indicating potential success with volume-based pricing strategies.

These observations suggest that pricing strategies should consider a balance between unit price and total quantity ordered to optimize total revenue.